compiler-cli/src/config.rs

```diff
@@ -55,12 +55,14 @@ pub fn find_package_config_for_module(
 
 pub fn read(config_path: PathBuf) -> Result<PackageConfig, Error> {
     let toml = crate::fs::read(&config_path)?;
-    toml::from_str(&toml).map_err(|e| Error::FileIo {
+    let config: PackageConfig = toml::from_str(&toml).map_err(|e| Error::FileIo {
         action: FileIoAction::Parse,
         kind: FileKind::File,
         path: config_path,
         err: Some(e.to_string()),
-    })
+    })?;
+    config.check_gleam_compatability()?;
```

Comment:
I'd like to be able to test that if you run (or perhaps instantiate) the package compiler with a version too low it errors, to ensure this behaviour isn't lost in future refactorings. Could we move it there and have a test? 🙏 

Anything that isn't going to be obviously wrong with normal manual testing it helps to have tests for. We accidentally lost the printing of warnings once 😁

Reply:
Currently there are two checks: one in the cli and one in the package compiler. The cli check covers stuff for the root package, and gives a nice error when for things like `deps download` that don't build anything. Do you want me to remove this check?

Reply:
Ah! That makes more sense, thank you. Keeping and having a test for the core one sounds good.